src/tiled/command.cpp

```diff
@@ -50,6 +51,46 @@ QString Command::finalCommand() const
     return finalCommand;
 }
 
+QString Command::finalWorkingDir() const
+{
+    QString finalWorkingDir = workingDir;
+
+    MapDocument *mapDoc = DocumentManager::instance()->currentDocument();
+    if (mapDoc) {
+        const QString mapPath = mapDoc->path();
+        finalWorkingDir.replace(QLatin1String("%executablepath"),
+                                QLatin1String("\"%1\"")).arg(executablePath());
+        finalWorkingDir.replace(QLatin1String("%mappath"),
+                                QLatin1String("\"%1\"")).arg(mapPath);
```

Comment:
Hmm, did you actually test this? I tried out your patch and couldn't get it to work, and it's because this code doesn't actually put any executable or map path into the final working directory. The `QString::arg` function returns a copy rather than doing an in-place replacement.

I think you shouldn't use `QString::arg` here anyway. One problem is that when for example no `%executablepath` is present in the working directory, `arg` will complain to standard error about a missing argument since it will be called on a string without `%1`. But also it is simply not needed. Just use `QString::replace` to replace the variables directly:

```cpp
        finalWorkingDir.replace(QLatin1String("%executablepath"), executablePath());
        finalWorkingDir.replace(QLatin1String("%mappath"), mapPath);
```